src/components/Panel.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { ReactNode } from 'react';
+
+export type PanelProps = {
+  children: ReactNode;
+  panelStyle?: 'rounded' | 'square';
+};
+
+export const Panel = ({ children, panelStyle = 'rounded' }: PanelProps) => (
+  <div className={`raf-panel raf-panel--${panelStyle}`}>{children}</div>
+);
+
+type ChildPanelProps = Omit<PanelProps, 'panelStyle'>;
+
+export type PanelContentProps = ChildPanelProps;
+export type PanelFooterProps = ChildPanelProps;
+export type PanelHeaderProps = ChildPanelProps;
+
+export const PanelContent = ({ children }: PanelContentProps) => (
+  <div className="raf-panel-content">{children}</div>
+);
+
+export const PanelFooter = ({ children }: PanelFooterProps) => (
+  <div className="raf-panel-footer">{children}</div>
+);
+
+export const PanelHeader = ({ children }: PanelHeaderProps) => (
+  <div className="raf-panel-header">{children}</div>
+);
```

Comment:
these 3 don't accept `panelStyle`

Reply:
It doesn't, there's an `Omit` at line 12.

![image](https://user-images.githubusercontent.com/43254280/109283613-490f7080-781f-11eb-95e2-de88ae03fb6a.png)

Reply:
why not simply using `{ children: ReactNode }`

Reply:
Just to make it look cooler. 😎 I mean, there's little to no difference (same LoC) and this way it "points" to the `PanelProps` to simulate inheritance of sorts if you know what I mean.

Reply:
haha simplicity is usually better but I see your point